firmware/vcv_plugin/internal/make_element.cc

```diff
@@ -100,6 +100,20 @@ static Knob create_base_knob(rack::app::Knob *widget) {
 		element.display_offset = pq->displayOffset;
 		element.integral = pq->snapEnabled;
 		element.display_precision = pq->displayPrecision;
+
+		if (element.integral) {
+			element.num_pos = pq->maxValue - pq->minValue + 1;
+
+			auto clamped_num_pos = std::min<size_t>(element.num_pos, pq->labels.size());
+
+			if (clamped_num_pos < element.num_pos && element.num_pos > 0) {
+				pr_warn("Warning: Snapped knob has %d positions, but only %d labels\n", element.num_pos, clamped_num_pos);
+			}
+
+			for (auto i = 0u; i < clamped_num_pos; i++) {
+				element.pos_names[i] = pq->labels[i];
+			}
```

Comment:
Here is where mscHack crashes.
`pq->labels.size()` is 25, and so is `pq->maxValue - pq->minValue + 1`.

But element.pos_names.size() is 16, so it crashes when we write out of bounds.

Reply:
Also the warning check should just tell us if we truncated the list of labels:

```c++
	if (clamped_num_pos < pq->labels.size()) {
		pr_warn("Warning: Snapped knob has %u labels, but only %u were used\n", pq->labels.size(), clamped_num_pos);
	}

```